### PR TITLE
add ability to change compression codec. This allows us to use DEFLATE instead of lz4.

### PR DESCRIPTION
**Pull request by andrewthad**
…E instead of lz4.

### CODE REVIEW

**Comment by bitemyapp** [comment]
Are you trying to save CPU time? I'm curious what triggered this.

**Comment by andrewthad** [comment]
Other way around. I'm trying to save disk save. According to the [disk tuning docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-disk-usage.html#_use_literal_best_compression_literal):

> The _source and stored fields can easily take a non negligible amount of disk space. They can be compressed more aggressively by using the best_compression codec.

File: src/Database/V5/Bloodhound/Types.hs [hunk 1]
```diff
@@ -600,6 +602,20 @@ data ReplicaBounds = ReplicasBounded Int Int
                    | ReplicasUnbounded
                    deriving (Eq, Read, Show, Generic, Typeable)
 
+data Compression = CompressionDefault | CompressionBest
```
Comment (bitemyapp) [inline src/Database/V5/Bloodhound/Types.hs]: I realize these data constructors are representative of the underlying text in the JSON but I'd like some documentation of what compression algorithm they are likely to force in the Lucene index and why you'd want one versus the other.

**Comment by andrewthad** [comment]
I've improved the docs. Also, I forgot to mention that the feature added in this PR cannot be used without first merging https://github.com/bitemyapp/bloodhound/pull/209. This index setting isn't cannot actually be updated dynamically, so it needs `createIndexWith` from the other PR.

**Comment by bitemyapp** [comment]
@andrewthad could you add the compression setting to the `createIndexWith` test case? If Elastic changes the API I'd like the tests to go red.

**Comment by andrewthad** [comment]
when has elastic ever made haphazardly documented breaking changes to their api? ;)
I'll close this PR and rebase on top of https://github.com/bitemyapp/bloodhound/pull/209 and then add the tests there.

**Comment by andrewthad** [comment]
Ah, you just merged it. Nevermind, I'll just rebase this on top of master and add the test.

**Comment by bitemyapp** [comment]
Gah, sorry if I'm making this difficult for you!

**Comment by andrewthad** [comment]
No worries. There's now a test for both compression codecs.

**Comment by bitemyapp** [comment]
@andrewthad okay, should I merge this one then?

**Comment by andrewthad** [comment]
Yes, this one's got everything now.

**Comment by bitemyapp** [comment]
@andrewthad thank you for everything!

**Pull request merged**